flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultExecutionVertexOperations.java

```diff
@@ -35,4 +35,9 @@ public void deploy(final ExecutionVertex executionVertex) throws JobException {
 	public CompletableFuture<?> cancel(final ExecutionVertex executionVertex) {
 		return executionVertex.cancel();
 	}
+
+	@Override
+	public void markFailed(final ExecutionVertex executionVertex, final Throwable cause) {
+		executionVertex.markFailed(cause);
```

Comment:
Solution looks correct. What is not so nice is that the meaning of the `fromSchedulerNg` is hard(er) to understand.

Reply:
Yes. And that's why I had wanted to do FLINK-14606 to use a `isInternalError` flag to supersede `fromSchedulerNg`.
I think we can simplify it after the legacy scheduler is removed.